standalone-metastore/metastore-common/src/main/java/org/apache/hadoop/hive/metastore/HiveMetaStoreClient.java

```diff
@@ -798,14 +798,14 @@ public void close() {
     // just in case, we make this call.
     if ((transport != null) && transport.isOpen()) {
       transport.close();
-      final int newCount = connCount.decrementAndGet();
-      LOG.info("Closed a connection to metastore, current connections: {}",
-          newCount);
       if (LOG.isTraceEnabled()) {
         LOG.trace("METASTORE CONNECTION TRACE - close [{}]",
             System.identityHashCode(this), new Exception());
       }
     }
+    final int newCount = connCount.decrementAndGet();
```

Comment:
this should be done only if client is not null at the first place. otherwise if you just keep calling close it will keep decrementing